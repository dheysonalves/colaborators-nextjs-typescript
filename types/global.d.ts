import '@emotion/react'

declare module '@emotion/react' {
	export interface Theme {
		light: {
			brand: {
				blue: string
			}
			neutral: {
				background: string
				light: string
				black: string
				neutral_1: string
				neutral_2: string
				neutral_3: string
				neutral_5: string
			}
			primary: {
				default: string
				green: string
			}
			feedback: {
				success: string
			}
			tertiary: {
				disabled: string
			}
		}
		dark: {
			brand: {
				blue: string
			}
			neutral: {
				background: string
				light: string
				black: string
				neutral_2: string
				neutral_3: string
				neutral_5: string
			}
			primary: {
				default: string
				green: string
			}
			feedback: {
				success: string
			}
			tertiary: {
				disabled: string
			}
		}
	}
}
