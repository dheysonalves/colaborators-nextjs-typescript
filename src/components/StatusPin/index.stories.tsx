import { ComponentMeta, ComponentStory } from '@storybook/react'
import StatusPin from '.'

export default {
	title: 'Composites/StatusPin',
	component: StatusPin,
} as ComponentMeta<typeof StatusPin>

export const Default: ComponentStory<typeof StatusPin> = () => <StatusPin />
