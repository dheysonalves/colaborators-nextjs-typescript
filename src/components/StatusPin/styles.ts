import styled from '@emotion/styled'
import { BaseTypography } from '../Typography/base.index'

export const Pin = styled.div`
	display: flex;
	justify-content: center;
	border-radius: 80px;
	background-color: ${(props) =>
		props.color === 'active'
			? props.theme.light.feedback.success
			: props.theme.light.tertiary.disabled};
	padding: 4px 8px;
	width: 72px;
	height: 24px;
`

export const StatusText = styled(BaseTypography)`
	font-size: 14px;
	color: ${(props) => props.theme.light.neutral.black};
	font-weight: 500;
	line-height: 16.24px;
`
