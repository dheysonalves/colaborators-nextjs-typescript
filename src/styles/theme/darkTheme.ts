export const theme = {
	dark: {
		brand: {
			blue: '#034AFD',
		},
		neutral: {
			background: '#F8FAF9',
			light: '#fff',
			black: '#34423D',
			neutral_2: '#F5FAF8',
			neutral_3: '#A3B8B0',
			neutral_5: '#587169',
		},
		primary: {
			default: '#22E0A1',
			green: '#B5F1DD',
		},
		feedback: {
			success: '#B5F1DD',
		},
		tertiary: {
			disabled: '#EAEFED',
		},
	},
}
