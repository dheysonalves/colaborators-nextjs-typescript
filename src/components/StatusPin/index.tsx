import React from 'react'
import { AgentStatusTypes } from '../../api/@types/general.types'

import { Pin, StatusText } from './styles'

type StatusPinProps = {
	status: AgentStatusTypes | string
}

const StatusPin = ({ status }: StatusPinProps) => {
	const text = status === 'active' ? 'Ativo' : 'Inativo'
	return (
		<Pin color={status}>
			<StatusText tag="h3">{text}</StatusText>
		</Pin>
	)
}

export default StatusPin
