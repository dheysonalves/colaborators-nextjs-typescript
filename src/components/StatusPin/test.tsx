import React from 'react'
import { render, screen, cleanup } from '@testing-library/react'
import { AgentStatusTypes } from '../../api/@types/general.types'
import { theme } from '../../styles/theme/theme'
import StatusPin from '.'
import { ThemeProvider } from '@emotion/react'

const setup = (status: AgentStatusTypes) => {
	const utils = render(
		<ThemeProvider theme={theme}>
			<StatusPin status={status} />
		</ThemeProvider>,
	)
	return {
		...utils,
	}
}

describe('<StatusPin />', () => {
	afterEach(cleanup)

	it('should render the heading Ativo', () => {
		const { container } = setup(AgentStatusTypes.ACTIVE)

		expect(screen.getByRole('heading', { name: /Ativo/i })).toBeInTheDocument()
		expect(container.firstChild).toHaveStyle({
			'background-color': '#B5F1DD',
		})

		expect(container.firstChild).toMatchSnapshot()
	})

	it('should render the heading Inativo', () => {
		const { container } = setup(AgentStatusTypes.INACTIVE)

		expect(
			screen.getByRole('heading', { name: /Inativo/i }),
		).toBeInTheDocument()

		expect(container.firstChild).toHaveStyle({
			'background-color': '#EAEFED',
		})

		expect(container.firstChild).toMatchSnapshot()
	})
})
