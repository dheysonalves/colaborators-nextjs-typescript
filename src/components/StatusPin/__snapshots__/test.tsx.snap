// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StatusPin /> should render the heading Ativo 1`] = `
<div
  class="css-1xl4ljt"
  color="active"
>
  <h3
    class="css-1rt2vi7"
  >
    Ativo
  </h3>
</div>
`

exports[`<StatusPin /> should render the heading Inativo 1`] = `
<div
  class="css-tvybe5"
  color="inactive"
>
  <h3
    class="css-1rt2vi7"
  >
    Inativo
  </h3>
</div>
`
